web/models/build_status_updater.ex

```diff
@@ -1,9 +1,27 @@
 defmodule Bigben.BuildStatusUpdater do
+  alias Bigben.Branch
   alias Bigben.Build
   alias Bigben.Repo
 
   def update(build_params) do
-    changeset = Build.changeset(%Build{}, build_params)
+    branch = get_branch(build_params)
+    updated_params = build_params
+      |> Map.delete(:branch)
+      |> Map.put("branch_id", branch.id)
+
+    changeset = Build.changeset(%Build{}, updated_params)
+    Repo.insert!(changeset)
+  end
+
+  def get_branch(%{"branch" => branch}) do
+    case Repo.get_by(Branch, name: branch) do
+      nil -> create_branch(branch)
+      branch -> branch
+    end
```

Comment:
I think this could be simplified to `Repo.get_by(Branch, name: branch) || create_branch(branch)`